doc/source/contributing.rst

```diff
@@ -0,0 +1,327 @@
+
+For Python 3.5, you can download and install the `Visual Studio 2015 Community Edition <https://www.visualstudio.com/en-us/downloads/visual-studio-2015-downloads-vs.aspx>`__.
+
+Here are some references:
+
+- https://github.com/conda/conda-recipes/wiki/Building-from-Source-on-Windows-32-bit-and-64-bit
+- https://cowboyprogrammer.org/building-python-wheels-for-windows/
+- https://blog.ionelmc.ro/2014/12/21/compiling-python-extensions-on-windows/
+- https://support.enthought.com/hc/en-us/articles/204469260-Building-Python-extensions-with-Canopy
+
+
+
+
+3) Installing Dependencies
+---------------------------
+
+To run *geopandas* in an development environment, you must first install *geopandas*'s dependencies. We suggest doing so using the following commands (executed after your development environment has been activated)::
+
+    conda install -c conda-forge fiona shapely pyproj rtree
+    conda install pandas
+
+
+This should install all necessary dependencies.
+
+4) Making a development build
+-------------------------------------
+
+Once dependencies are in place, make an in-place build by navigating to the git clone of the *geopandas* repository and running::
+
+    python setup.py develop
+
+
+
+
+5) Making changes and writing tests
+-------------------------------------
+
+*geopandas* is serious about testing and strongly encourages contributors to embrace
+`test-driven development (TDD) <http://en.wikipedia.org/wiki/Test-driven_development>`_.
+This development process "relies on the repetition of a very short development cycle:
+first the developer writes an (initially failing) automated test case that defines a desired
+improvement or new function, then produces the minimum amount of code to pass that test."
+So, before actually writing any code, you should write your tests.  Often the test can be
+taken from the original GitHub issue.  However, it is always worth considering additional
+use cases and writing corresponding tests.
+
+Adding tests is one of the most common requests after code is pushed to *geopandas*.  Therefore,
+it is worth getting in the habit of writing tests ahead of time so this is never an issue.
+
+Like many packages, *geopandas* uses the `Nose testing system
+<http://nose.readthedocs.org/en/latest/index.html>`_ and the convenient
+extensions in `numpy.testing
+<http://docs.scipy.org/doc/numpy/reference/routines.testing.html>`_.
+
+Writing tests
+~~~~~~~~~~~~~
+
+All tests should go into the ``tests`` subdirectory of the specific package.
+This folder contains many current examples of tests, and we suggest looking to these for
+inspiration.
+
+The ``.util`` module has many special ``assert`` functions that
+make it easier to make statements about whether Series or DataFrame objects are
+equivalent. The easiest way to verify that your code is correct is to
+explicitly construct the result you expect, then compare the actual result to
+the expected correct result::
```

Comment:
This should probably give an example with the actual geopandas functions (eg `assert_geoseries_equal`)